### PR TITLE
Add deregister user documentation

### PR DESCRIPTION
**Pull request by LGLO**
# Description

Adds 'deregistration' user-guide

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages.
- [x] New tests are added if needed and existing tests are updated.
- [x] Relevant logging and metrics added
- [x] CI passes. See note on CI.
- [x] Any changes are noted in the `changelog.md` for affected crate
- [x] Self-reviewed the diff

### CODE REVIEW

File: docs/user-guides/registered.md [hunk 1]
```diff
@@ -334,6 +335,17 @@ The wizard sets the required environment variables and starts the node.
 
 Registration is effective after 1-2 Cardano epochs. After the waiting period, the partner chain node is registered on the partner chain and is a selection option for the consensus committee.
 
+### 7. Optionally: deregister from the partner chain
+
+To deregister from the block producers candidates, you need to run the deregister wizard.
```
Comment (nstanford5) [inline docs/user-guides/registered.md:340]:
```suggestion
To deregister from the list of block producer candidates, you need to run the deregister wizard.
```

File: docs/user-guides/registered.md [hunk 2]
```diff
@@ -334,6 +335,17 @@ The wizard sets the required environment variables and starts the node.
 
 Registration is effective after 1-2 Cardano epochs. After the waiting period, the partner chain node is registered on the partner chain and is a selection option for the consensus committee.
 
+### 7. Optionally: deregister from the partner chain
```
Comment (nstanford5) [inline docs/user-guides/registered.md:338]:
```suggestion
### 7. Optional: deregister from the partner chain
```

File: docs/user-guides/registered.md [hunk 3]
```diff
@@ -14,6 +14,7 @@ A registered block producer is a Cardano stake pool operator (SPO) that desires
 4. Obtain chain parameters from the chain builder
 5. Register for the partner chain
 6. Run the partner chain node
+7. Optionally: deregister from the partner chain
```
Comment (nstanford5) [inline docs/user-guides/registered.md:17]:
```suggestion
7. Optional: deregister from the partner chain
```

**Review by nstanford5 — APPROVED** [review]

**Review by Lupowilk — APPROVED** [review]
LGTM, just resolve the few minor things pointed out by Nick.

**Pull request merged**